alf/networks/inverse_mvp_network.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2020 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""InverseMVPNetwork"""
+
+import gin
+import functools
+import math
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import alf.utils.math_ops as math_ops
+import alf.nest as nest
+from alf.initializers import variance_scaling_init
+from alf.tensor_specs import TensorSpec
+
+from .network import Network
+from .encoding_networks import EncodingNetwork
+from .projection_networks import NormalProjectionNetwork
+
+
+@gin.configurable
+class InverseMVPNetwork(Network):
```

Comment:
InverseMVPNetwork can be constructed using EncodingNetwork easily as the following. So this Network does not add much value. Creating the model directly using EncodingNetwork also make the code more understandable as it can be easily figure out what it does (assuming knowing EnodingNetwork).
```python
EncodingNetwork(
    input_tensor_spec,
    input_prepreprocessors=(
        Linear(z_dim, hidden_size),
        Linear(eps_dim, hidden_size)),
    preprocessing_combiner=layers.NestConcat()),
    fc_layer_params=(2 * hidden_size, ) * num_hidden_layers,
    activation=activation,
    kernel_initializer=kernel_initializer,
    last_layer_size=output_dim,
    last_activation=math_ops.identity
)
```
Instead of making new InverseMVPNetwork class, you can add a helper function create_mvp_network() in generator.py to do the above work.

Reply:
agreed